### PR TITLE
Added an automatic class occurrence counter for Focal Loss

### PR DESCRIPTION
**Pull request by SuzannaLin**
Hi Yassine,
I finally got to writing a class occurrence counter for the supervised training samples. I added comments to explain what I did.

Best regards,
Suzanna

### CODE REVIEW

**Review by yassouali — COMMENTED** [review]
First of all, thank you for the pull request, I just have some remarks, maybe if you have time to check them.

**Review by yassouali — CHANGES_REQUESTED** [review]
can you make this into a function in losses.py (just to keep train.py as short as possible).

For instance: 
```python
from losses import get_FL_loss

sup_loss = get_FL_loss(supervised_loader, num_classes)
```

Also, I think the number of classes needs to be fetched from the config or dataset, and not set to 4.

Thank you!

**Comment by SuzannaLin** [comment]
Thanks, Yassine! I will do my best to add those changes!

**Comment by SuzannaLin** [comment]
I tried to resolve the conflict. I think it should be good now.
Can you let me know if there is anything else that should be changed?

**Comment by yassouali** [comment]
Thank you very much @SuzannaLin
